Formula/gz-cmake4.rb

```diff
@@ -0,0 +1,44 @@
+class GzCmake4 < Formula
+  desc "CMake helper functions for building robotic applications"
+  homepage "https://gazebosim.org"
+  url "https://github.com/gazebosim/gz-cmake.git", branch: "main"
+  version "3.999.999~0~20231006"
```

Comment:
I believe `brew` doesn't like `~` in version strings anymore (see https://github.com/osrf/homebrew-simulation/issues/2390 and https://github.com/gazebo-tooling/release-tools/pull/1007)

I think I need to update the tool used to generate these pull requests

Reply:
https://github.com/gazebo-tooling/release-tools/pull/1028